src/meta/src/stream/source_manager.rs

```diff
@@ -495,6 +497,32 @@ impl SourceManagerCore {
                 self.actor_splits.remove(actor_id);
             }
         }
+
+        if let Some(fragment_replacements) = fragment_replacements {
+            for (old_fragment_id, new_fragment_id) in fragment_replacements {
+                for fragment_ids in self.source_fragments.values_mut() {
+                    if fragment_ids.remove(&old_fragment_id) {
+                        fragment_ids.insert(new_fragment_id);
+                    }
```

Comment:
Why didn't we have this step prior to this PR, considering we've already supported schema change on table with source?

Reply:
I had the same question. I think it's overlooked, i.e., we forgot to GC old fragment here. New fragment will be added by `added_source_fragments` above.